crates/proto/src/error.rs

```diff
@@ -440,6 +446,28 @@ impl ProtoError {
 
                     let response = response;
                     let soa = response.soa().as_ref().map(RecordRef::to_owned);
+
+                    // Collect any referral nameservers and associated glue records
+                    let mut referral_name_servers = vec![];
+
+                    for ns in response.name_servers().iter() {
+                        let glue: Vec<Record> = response
+                            .additionals()
+                            .iter()
+                            .filter(|record| *record.name() == ns.data().as_ns().unwrap().0)
+                            .filter(|record| record.data().as_a().is_some() || record.data().as_aaaa().is_some())
+                            .map(Record::to_owned)
```

Comment:
Why is the `unwrap()` safe? I'd prefer a single `filter_map()` over `filter().filter().map()`.